src/OversetSimulation.cpp

```diff
@@ -143,4 +149,45 @@ bool OversetSimulation::do_connectivity(const int tstep)
     return (tstep > 0) && (tstep % m_overset_update_interval) == 0;
 }
 
+long OversetSimulation::mem_usage_all(const int step)
+{
+    struct rusage usage;
```

Comment:
This has a lot in common with the other call in ExawindSolver. Can we maybe consolidate the memory stuff in a separate file? and then put the ifdefs there? That way we we ever want to update the memory stuff for other platforms we can just look at that file.